public/js/components/Frames.js

```diff
@@ -66,6 +66,9 @@ const Frames = React.createClass({
 
   renderFrames() {
     let { frames } = this.props;
+    if (!frames) {
+      return null;
+    }
```

Comment:
Oh, actually `frames` is a `Seq`, not an array.  I guess we can do without `frames = []`.

But what do you think of the rest of the function cleanup?